atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/InstrumentedCassandraClient.java

```diff
@@ -0,0 +1,160 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.cassandra.thrift.AutoDelegate_Client;
+import org.apache.cassandra.thrift.Cassandra;
+import org.apache.cassandra.thrift.ColumnOrSuperColumn;
+import org.apache.cassandra.thrift.ColumnParent;
+import org.apache.cassandra.thrift.ConsistencyLevel;
+import org.apache.cassandra.thrift.InvalidRequestException;
+import org.apache.cassandra.thrift.KeyRange;
+import org.apache.cassandra.thrift.KeySlice;
+import org.apache.cassandra.thrift.Mutation;
+import org.apache.cassandra.thrift.SlicePredicate;
+import org.apache.cassandra.thrift.TimedOutException;
+import org.apache.cassandra.thrift.UnavailableException;
+import org.apache.thrift.TException;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.palantir.atlasdb.logging.LoggingArgs;
+import com.palantir.atlasdb.tracing.CloseableTrace;
+import com.palantir.atlasdb.util.AtlasDbMetrics;
+import com.palantir.processors.AutoDelegate;
+
+@AutoDelegate(typeToExtend = Cassandra.Client.class)
+@SuppressWarnings({"all"}) // thrift variable names
+public class InstrumentedCassandraClient extends AutoDelegate_Client {
+    private static final String SERVICE_NAME = "cassandra-thrift-client";
+    private static final MetricRegistry METRIC_REGISTRY = AtlasDbMetrics.getMetricRegistry();
+
+    private static final Timer MULTIGET_SLICE_TIMER = METRIC_REGISTRY.timer(
+            MetricRegistry.name(InstrumentedCassandraClient.class, "multiget_slice"));
+    private static final Timer BATCH_MUTATE_TIMER = METRIC_REGISTRY.timer(
+            MetricRegistry.name(InstrumentedCassandraClient.class, "batch_mutate"));
+    private static final Timer GET_RANGE_SLICE_TIMER = METRIC_REGISTRY.timer(
+            MetricRegistry.name(InstrumentedCassandraClient.class, "get_range"));
+
+    private Cassandra.Client delegate;
+
+    public InstrumentedCassandraClient(Cassandra.Client delegate) {
+        super(delegate.getInputProtocol());
+        this.delegate = delegate;
+    }
+
+    @Override
+    public Cassandra.Client delegate() {
+        return delegate;
+    }
+
+    @Override
+    public void batch_mutate(Map<ByteBuffer, Map<String, List<Mutation>>> mutation_map,
+            ConsistencyLevel consistency_level)
+            throws InvalidRequestException, UnavailableException, TimedOutException, TException {
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("client.batch_mutate(number of rows {}, consistency {})",
+                mutation_map.size(), toString(consistency_level))) {
+            registerDuration(() -> {
```

Comment:
Any reason not to use tritium for these? I guess we'd need to extract an interface for the Client.. though at this point I'm pretty surprised we haven't decided to do that already :)